sdks/java/core/src/main/java/org/apache/beam/sdk/util/state/StateTags.java

```diff
@@ -61,11 +60,28 @@ private StateTags() { }
     StateTag<K, StateT> asKind(StateKind kind);
   }
 
+  /** Create a state tag for the given id and spec. */
+  public static <K, StateT extends State> StateTag<K, StateT> tagForSpec(
+      String id, StateSpec<K, StateT> spec) {
+    return new SimpleStateTag<>(new StructuredId(id), spec);
+  }
+
+  /**
+   * Quirk of type system unresolved. TODO: make this and the above method one method.
+   */
+  public static <K, InputT, AccumT, OutputT>
+  StateTag<? super K, AccumulatorCombiningState<InputT, AccumT, OutputT>> tagforSpec(
```

Comment:
This should be capitolized the same as `tagForSpec`, otherwise the difference should be explained.